app/assets/javascripts/libs/datasource.schema.json

```diff
@@ -0,0 +1,137 @@
+            "properties": {
+              "name": { "type": "string" },
+              "resolutions": { "type": "array", "items": { "type": "number" } },
+              "boundingBox": { "$ref": "#/definitions/types::BoundingBox" }
+            },
+            "required": ["name", "resolutions", "boundingBox"]
+          }
+        }
+      },
+      "required": ["dataFormat", "sections"]
+    },
+    "types::DataLayer": {
+      "title": "DataLayer",
+      "allOf": [
+        {
+          "title": "DataLayerBasic",
+          "type": "object",
+          "properties": {
+            "name": { "type": "string" },
+            "category": { "enum": ["color", "segmentation"] },
+            "elementClass": { "enum": ["uint8", "uint16", "uint24", "uint32", "uint64"] }
+          },
+          "required": ["name", "category", "elementClass"]
+        },
+        {
+          "title": "DataLayerCategories",
+          "anyOf": [
+            {
+              "title": "DataLayerColor",
+              "type": "object",
+              "properties": { "category": { "const": "color" } },
+              "required": ["category"]
+            },
+            {
+              "title": "DataLayerSegmentation",
+              "type": "object",
+              "properties": {
+                "category": { "const": "segmentation" },
+                "largestSegmentId": { "type": "number", "minimum": 1 },
+                "mappings": { "type": "array", "items": { "type": "string" } }
+              },
+              "required": ["category", "largestSegmentId"]
+            }
+          ]
+        },
+        {
+          "title": "DataLayerFormats",
+          "anyOf": [
+            { "$ref": "#/definitions/types::DataLayerWKWPartial" },
+            { "$ref": "#/definitions/types::DataLayerKnossosPartial" }
+          ]
+        }
+      ]
+    },
+    "types::DatasourceConfiguration": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "object",
+          "properties": { "name": { "type": "string" }, "team": { "type": "string" } },
+          "additionalProperties": false,
+          "required": ["name", "team"]
+        },
+        "dataLayers": { "type": "array", "items": { "$ref": "#/definitions/types::DataLayer" } },
+        "scale": {
+          "type": "array",
+          "items": [
+            { "type": "number", "minimum": 0, "exclusiveMinimum": true },
+            { "type": "number", "minimum": 0, "exclusiveMinimum": true },
+            { "type": "number", "minimum": 0, "exclusiveMinimum": true }
+          ]
+        }
+      },
+      "required": ["id", "dataLayers", "scale"]
+    },
+    "types::LayerUserConfiguration": {
+      "type": "object",
+      "additionalProperties": {
+        "type": "object",
+        "properties": {
+          "brightness": { "type": "number" },
+          "contrast": { "type": "number" },
+          "color": { "$ref": "#/definitions/types::Vector3" }
```

Comment:
The Vector3 definition doesn't seem to work as intended, during testing I noticed that as long as the array contains only numbers (but more or less than 3), the validation is satisfied - not sure whether this can be fixed easily.